src/replica/replication_app_base.cpp

```diff
@@ -177,18 +194,32 @@ error_code replica_app_info::store(const char *file)
         marshall(writer, tmp, DSF_THRIFT_JSON);
     }
 
-    std::string ffile = std::string(file);
+    std::string ffile(file);
     std::string tmp_file = ffile + ".tmp";
 
-    std::ofstream os(tmp_file.c_str(),
-                     (std::ofstream::out | std::ios::binary | std::ofstream::trunc));
-    ERR_LOG_AND_RETURN_NOT_TRUE(
-        os.is_open(), ERR_FILE_OPERATION_FAILED, "open file {} failed", tmp_file);
-
-    auto data = writer.get_buffer();
-    os.write((const char *)data.data(), (std::streamsize)data.length());
-    os.close();
-
+    disk_file *hfile = file::open(tmp_file.c_str(), O_WRONLY | O_CREAT | O_BINARY | O_TRUNC, 0666);
```

Comment:
I find a lot of same code between L106~L130 and L200~L222, so could you refactor it by encapsulating them into a standalone function?